resolver/mindeps/__main__.py

```diff
@@ -91,39 +92,43 @@ def main_parser() -> argparse.ArgumentParser:
     return parser
 
 
-def task() -> None:  # noqa: C901
-    parser = main_parser()
-    args = parser.parse_args()
+class MinimumDependencyProvider(resolver.Provider):
+    def sort_candidates(
+        self,
+        candidates: Iterable[resolver.Candidate],
+    ) -> Sequence[resolver.Candidate]:
+        return sorted(candidates, key=operator.attrgetter('version'), reverse=False)
 
-    if args.requirements:
-        for bad_arg in _MARKER_KEYS + ('extras',):
-            if bad_arg in args:
-                resolver.__main__._error(f'Option --{bad_arg} not supported when specifying bare requirements')
 
-    reporter = resolver.__main__.VerboseReporter if args.verbose else resolvelib.BaseReporter
+def get_min_deps(
+    requirements: List[str],
+    reporter: Optional[resolvelib.BaseReporter] = None,
+    extras: Optional[Set[str]] = None,
+    markers: Optional[Dict[str, str]] = None
+) -> Dict[str, str]:
+    reporter = reporter or resolvelib.BaseReporter
     package_resolver = resolvelib.Resolver(
-        resolver.mindeps.MinimumDependencyProvider(
+        MinimumDependencyProvider(
             '/tmp/resolver-cache' if os.name == 'posix' else None
         ),
         reporter(),
     )
 
     requirements: Iterable[packaging.requirements.Requirement] = map(
-        packaging.requirements.Requirement,
-        args.requirements or _project_requirements(),
+        packaging.requirements.Requirement, requirements
     )
 
-    extras = set(vars(args).get('extras', {})) | {''}
-    if any(arg in _MARKER_KEYS for arg in vars(args)):
+    extras = extras.copy() | {''} if extras else {''}
+    if markers is not None and any(marker in _MARKER_KEYS for marker in markers):
```

Comment:
nitpick: I'd skip the `is None`, to simplify the condition.

```suggestion
    if markers and any(marker in _MARKER_KEYS for marker in markers):
```